### PR TITLE
feat (api): workflow run, add build variable

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: sdk/messages.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ var (
 	MsgSpawnInfoWorkerEnd                  = &Message{"MsgSpawnInfoWorkerEnd", trad{FR: "Le worker %s a terminé et a passé %s à travailler sur les étapes", EN: "Worker %s finished working on this job and took %s to work on the steps"}, nil}
 	MsgSpawnInfoJobTaken                   = &Message{"MsgSpawnInfoJobTaken", trad{FR: "Le job a été pris par le worker %s", EN: "Job was taken by worker %s"}, nil}
 	MsgSpawnInfoWorkerForJob               = &Message{"MsgSpawnInfoWorkerForJob", trad{FR: "Ce worker %s a été créé pour lancer ce job", EN: "This worker %s was created to take this action"}, nil}
+	MsgSpawnInfoJobError                   = &Message{"MsgSpawnInfoJobError", trad{FR: "Impossible de lancer ce job: %s", EN: "Unable to run thos job: %s"}, nil}
```
Comment (yesnault) [inline sdk/messages.go]: this or those.

File: sdk/workflow.go [hunk 2]
```diff
@@ -357,24 +357,25 @@ type WorkflowRun struct {
...
+	Stages             []Stage                   `json:"stages" db:"-"`
+	Start              time.Time                 `json:"start" db:"start"`
+	LastModified       time.Time                 `json:"last_modified" db:"last_modified"`
+	Done               time.Time                 `json:"done" db:"done"`
+	HookEvent          *WorkflowNodeRunHookEvent `json:"hook_event" db:"-"`
+	Manual             *WorkflowNodeRunManual    `json:"manual" db:"-"`
+	SourceNodeRuns     []int64                   `json:"source_node_runs" db:"-"`
+	Payload            []Parameter               `json:"payload" db:"-"`
+	PipelineParameters []Parameter               `json:"pipeline_parameter" db:"-"`
+	BuildParameters    []Parameter               `json:"build_parameter" db:"-"`
```
Comment (sguiheux) [inline sdk/workflow.go]: build_parameters?

File: engine/api/workflow_queue.go [hunk 3]
```diff
@@ -351,6 +351,49 @@ func postWorkflowJobTestsResultsHandler(w http.ResponseWriter, r *http.Request,
...
+
+	tx, errb := db.Begin()
+	if errb != nil {
+		return sdk.WrapError(errb, "postWorkflowJobVariableHandler> Unable to start tx")
+	}
+	defer tx.Rollback()
+
+	job, errj := workflow.LoadAndLockNodeJobRun(tx, id)
+	if errj != nil {
+		return sdk.WrapError(errr, "postWorkflowJobVariableHandler> Unable to load job")
```
Comment (yesnault) [inline engine/api/workflow_queue.go]: errj

File: engine/api/workflow_queue.go [hunk 4]
```diff
@@ -351,6 +351,49 @@ func postWorkflowJobTestsResultsHandler(w http.ResponseWriter, r *http.Request,
...
+	}
+
+	sdk.AddParameter(&job.Parameters, "cds.build."+v.Name, sdk.StringParameter, v.Value)
+
+	if err := workflow.UpdateNodeJobRun(tx, job); err != nil {
```
Comment (yesnault) [inline engine/api/workflow_queue.go]: errr -> err

File: engine/api/workflow_queue.go [hunk 5]
```diff
@@ -351,6 +351,49 @@ func postWorkflowJobTestsResultsHandler(w http.ResponseWriter, r *http.Request,
...
+
+	node, errn := workflow.LoadNodeRunByID(tx, job.WorkflowNodeRunID)
+	if errn != nil {
+		return sdk.WrapError(errn, "postWorkflowJobVariableHandler> Unable to load node")
+	}
+
+	sdk.AddParameter(&node.BuildParameters, "cds.build."+v.Name, sdk.StringParameter, v.Value)
+
+	if err := workflow.UpdateNodeRun(tx, node); err != nil {
+		return sdk.WrapError(errr, "postWorkflowJobVariableHandler> Unable to update node run")
```
Comment (yesnault) [inline engine/api/workflow_queue.go]: errr -> err

File: engine/api/workflow_queue.go [hunk 6]
```diff
@@ -351,6 +351,49 @@ func postWorkflowJobTestsResultsHandler(w http.ResponseWriter, r *http.Request,
...
+	}
+
+	if err := tx.Commit(); err != nil {
+		return sdk.WrapError(errr, "postWorkflowJobVariableHandler> Unable to commit tx")
```
Comment (yesnault) [inline engine/api/workflow_queue.go]: errr -> err

File: sdk/messages.go [hunk 7]
```diff
@@ -56,6 +56,7 @@ var (
 	MsgSpawnInfoWorkerEnd                  = &Message{"MsgSpawnInfoWorkerEnd", trad{FR: "Le worker %s a terminé et a passé %s à travailler sur les étapes", EN: "Worker %s finished working on this job and took %s to work on the steps"}, nil}
 	MsgSpawnInfoJobTaken                   = &Message{"MsgSpawnInfoJobTaken", trad{FR: "Le job a été pris par le worker %s", EN: "Job was taken by worker %s"}, nil}
 	MsgSpawnInfoWorkerForJob               = &Message{"MsgSpawnInfoWorkerForJob", trad{FR: "Ce worker %s a été créé pour lancer ce job", EN: "This worker %s was created to take this action"}, nil}
+	MsgSpawnInfoJobError                   = &Message{"MsgSpawnInfoJobError", trad{FR: "Impossible de lancer ce job: %s", EN: "Unable to run this job: %s"}, nil}
```
Comment (yesnault) [inline sdk/messages.go]: space

**Pull request closed without merging**